src/huggingface_hub/file_download.py

```diff
@@ -689,6 +706,61 @@ def _resumable_file_manager() -> "io.BufferedWriter":
     return cache_path
 
 
+def _normalize_etag(etag: str) -> str:
+    """Normalize ETag HTTP header, so it can be used to create nice filepaths.
+
+    The HTTP spec allows two forms of ETag:
+      ETag: W/"<etag_value>"
+      ETag: "<etag_value>"
+
+    The hf.co hub guarantees to only send the second form.
+
+    Args:
+        etag (str): HTTP header
+
+    Returns:
+        str: string that can be used as a nice directory name.
+    """
+    return etag.strip('"')
+
+
+def _create_relative_symlink(src: str, dst: str) -> None:
+    """Create a symbolic link named dst pointing to src as a relative path to dst.
+
+    The relative part is mostly because it seems more elegant to the author.
+
+    The result layout looks something like
+        └── [ 128]  snapshots
+            ├── [ 128]  2439f60ef33a0d46d85da5001d52aeda5b00ce9f
+            │   ├── [  52]  README.md -> ../../blobs/d7edf6bd2a681fb0175f7735299831ee1b22b812
+            │   └── [  76]  pytorch_model.bin -> ../../blobs/403450e234d65943a7dcf7e05a771ce3c92faa84dd07db4ac20f592037a1e4bd
```

Comment:
Showcasing this is elegant, and I think showcasing the full layout somewhere is warranted as it makes understanding it super simple. Happy to provide it!

Reply:
took a stab at it in 81b75678b3556544f6a493c110057c8746a41c78

Did that using `tree -h`